app/controllers/user_notifications_controller.rb

```diff
@@ -21,4 +21,28 @@ def delete
     ).destroy!
     render nothing: true
   end
+
+  def unsubscribe
+    user_notif = UserNotification.find_by(token: params[:token])
+    if user_notif.present?
+      Notification.find_each do |n|
+        notif = UserNotification.find_by(
```

Comment:
Can you come up with better variable names than `n`, `notif`, `user_notif`?

By itself usually these variables are not a problem. When they are mixed together, though, they can be confusing to differentiate.